source/MaterialXGraphEditor/Main.cpp

```diff
@@ -39,6 +40,38 @@ mx::FileSearchPath getDefaultSearchPath()
     return searchPath;
 }
 
+std::filesystem::path getConfigPath()
+{
+    std::filesystem::path config_dir;
+    auto xdg_config_home = mx::getEnviron("XDG_CONFIG_HOME");
+    auto home_directory = mx::getEnviron("HOME");
+    if (!xdg_config_home.empty())
+    {
+        config_dir = std::filesystem::path(xdg_config_home) / "MaterialX";
```

Comment:
This looks like a good proposal, but could it potentially be written in terms of the `FilePath` class in MaterialXFormat, rather than the `filesystem::path` class in C++17?

For now, we only _require_ C++11 in the MaterialX core libraries, though we support all C++ versions, because there are studios that still need to build MaterialX with GCC 5/6.

Since this new `getConfigPath` function has the potential to move to MaterialX core libraries in the future, it would be ideal to write it without adding any new dependencies.

Reply:
Yeah, that makes sense. It'll potentially be a little clunkier (since filesystem does a few more things than mkdir does) but that should be fine for this simple a case.

I'll get an update to the PR later this week

Reply:
@jstone-lucasfilm Updated with the use of FilePath so it should be portable to other parts of MaterialX now